manifests/service.pp

```diff
@@ -1,8 +1,43 @@
-# @summary A short summary of the purpose of this class
+# @summary Install InfluxDB package and manage service
 #
-# A description of what this class does
+# Install InfluxDB package and manage service
+#
+# @param manage_service
+#   Flag of whether to manage InfluxDB service
+#
+# @param required_packages
+#   List of package names to be installed (OS specific).
+#
+# @param service_name
+#   Name of InfluxDB service
 #
 # @example
 #   include profile_influxdb::service
-class profile_influxdb::service {
+class profile_influxdb::service (
+  Boolean       $manage_service,
+  Array[String] $required_packages,
+  String        $service_name,
+) {
+
+  # PACKAGES
+  ensure_packages( $required_packages, {'ensure' => 'present'} )
+
+  if ( $manage_service ) {
+    service { $service_name :
```

Comment:
This is ok as-is but was just curious. Is it anticipated that the service name might need customizing or it it setup as a variable here just to make it more like a template?

Reply:
I think I copied this logic from another service profile where the service name changes depending on what OS, etc is in play. It just seemed more extensible to leave it as a variable.